plugins/database/cassandra/cassandra_test.go

```diff
@@ -54,109 +59,152 @@ func TestCassandra_CreateUser(t *testing.T) {
+	password := "myreallysecurepassword"
+	createReq := newdbplugin.NewUserRequest{
+		UsernameConfig: newdbplugin.UsernameMetadata{
+			DisplayName: "test",
+			RoleName:    "test",
+		},
+		Statements: newdbplugin.Statements{
+			Commands: []string{createUserStatements},
+		},
+		Password:   password,
+		Expiration: time.Now().Add(1 * time.Minute),
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	createResp, err := db.NewUser(ctx, createReq)
 	if err != nil {
 		t.Fatalf("err: %s", err)
 	}
 
-	if err = testCredsExist(db.Hosts, db.Port, username, password); err != nil {
-		t.Fatalf("Could not connect with new credentials: %s", err)
+	assertCreds(t, db.Hosts, db.Port, createResp.Username, password, 5*time.Second)
+
+	deleteReq := newdbplugin.DeleteUserRequest{
+		Username: createResp.Username,
 	}
 
-	// Test default revoke statements
-	err = db.RevokeUser(context.Background(), statements, username)
+	ctx, cancel = context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	_, err = db.DeleteUser(context.Background(), deleteReq)
 	if err != nil {
 		t.Fatalf("err: %s", err)
 	}
 
-	if err = testCredsExist(db.Hosts, db.Port, username, password); err == nil {
-		t.Fatal("Credentials were not revoked")
-	}
+	assertNoCreds(t, db.Hosts, db.Port, createResp.Username, password)
 }
 
-func TestCassandra_RotateRootCredentials(t *testing.T) {
-	db, cleanup := getCassandra(t, 4)
-	defer cleanup()
-
-	if !db.cassandraConnectionProducer.Initialized {
-		t.Fatal("Database should be initialized")
+func assertCreds(t testing.TB, address string, port int, username, password string, timeout time.Duration) {
+	t.Helper()
+	op := func() error {
+		return connect(t, address, port, username, password)
 	}
+	bo := backoff.NewExponentialBackOff()
+	bo.MaxElapsedTime = timeout
+	bo.InitialInterval = 500 * time.Millisecond
+	bo.MaxInterval = bo.InitialInterval
+	bo.RandomizationFactor = 0.0
 
-	newConf, err := db.RotateRootCredentials(context.Background(), nil)
+	err := backoff.Retry(op, bo)
 	if err != nil {
-		t.Fatalf("err: %v", err)
+		t.Fatalf("failed to connect after %s: %s", timeout, err)
 	}
-	if newConf["password"] == "cassandra" {
-		t.Fatal("password was not updated")
+}
+
+func connect(t testing.TB, address string, port int, username, password string) error {
+	t.Helper()
+	clusterConfig := gocql.NewCluster(address)
+	clusterConfig.Authenticator = gocql.PasswordAuthenticator{
+		Username: username,
+		Password: password,
 	}
+	clusterConfig.ProtoVersion = 4
+	clusterConfig.Port = port
 
-	err = db.Close()
+	session, err := clusterConfig.CreateSession()
 	if err != nil {
-		t.Fatalf("err: %s", err)
+		return err
 	}
+	defer session.Close()
+	return nil
 }
 
-func testCredsExist(address string, port int, username, password string) error {
+func assertNoCreds(t testing.TB, address string, port int, username, password string) {
```

Comment:
I'm curious why `assertCreds()` needs a retry with backoff, but `assertNoCreds()` doesn't.

Reply:
Because I neglected to put it in `assertNoCreds`.